packages/central-server/src/apiV2/utilities/emailVerification.js

```diff
@@ -59,5 +59,16 @@ export const verifyEmailHelper = async (models, searchCondition, token) => {
     verified_email: searchCondition,
   });
 
-  return users.find(x => encryptPassword(x.email + x.password_hash, x.password_salt) === token);
+  for (const user of users) {
+    const verified = await verifyPassword(
+      `${user.email}${user.password_hash}`,
+      user.password_salt,
+      token,
+    );
```

Comment:
Argon2 including salt generation is also why this fails: this operation is no longer deterministic. You shouldn't reuse salts anyway so a good opportunity to fix this differently.